pennylane/_device.py

```diff
@@ -721,16 +716,48 @@ def batch_transform(self, circuit):
         hamiltonian_in_obs = "Hamiltonian" in [obs.name for obs in circuit.observables]
 
         if hamiltonian_in_obs and ((not supports_hamiltonian or finite_shots) or grouping_known):
+            # If the observable contains a Hamiltonian and the device does not
+            # support Hamiltonians, or if the simulation uses finite shots, or
+            # if the Hamiltonian explicitly specifies an observable grouping,
+            # split tape into multiple tapes of diagonalizable known observables.
             try:
-                return qml.transforms.hamiltonian_expand(circuit, group=False)
+                circuits, hamiltonian_fn = qml.transforms.hamiltonian_expand(circuit, group=False)
 
             except ValueError as e:
                 raise ValueError(
                     "Can only return the expectation of a single Hamiltonian observable"
                 ) from e
-
-        # otherwise, return an identity transform
-        return [circuit], lambda res: res[0]
+        else:
+            # otherwise, return the output of an identity transform
+            circuits, hamiltonian_fn = [circuit], lambda res: res[0]
+
+        # Check whether the circuit was broadcasted (then the Hamiltonian-expanded
+        # ones will be as well) and whether broadcasting is supported
+        batch_size = circuit.batch_size
+        supports_broadcasting = self.capabilities().get("supports_broadcasting")
+
+        if batch_size is None or supports_broadcasting:
+            # If the circuit wasn't broadcasted or broadcasting is supported, no action required
+            return circuits, hamiltonian_fn
+
+        unbroadcasted_circuits = []
+        unbroadcast_fns = []
+        # Expand each of the broadcasted Hamiltonian-expanded circuits
+        for c in circuits:
+            tapes_batch, fn = qml.transforms.unbroadcast_expand(c)
+            unbroadcasted_circuits.extend(tapes_batch)
+            unbroadcast_fns.append(fn)
```

Comment:
@dwierichs would

https://github.com/PennyLaneAI/pennylane/blob/bd552accef371b21bde3ed862a527ee140e31dfd/pennylane/transforms/batch_transform.py#L422

help avoid the need to manually loop over the circuit to apply the batch transform? It should allow you to just do

```python
return map_batch_transform(unbroadcast_expand, circuits)
```

Reply:
It was super useful! Reducing the addition to `Device.batch_transform` to a few lines :)